src/logger/index.ts

```diff
@@ -58,22 +72,31 @@ class ConsoleLogger {
     if (this.logLevel > LogLevel.debug) {
       return;
     }
-    this.logger.info(content, this.includeStackTrace);
+    this.log(LogStyle.info, content);
   }
 
   public imp(content: string | object): void {
     if (this.logLevel > LogLevel.important) {
       return;
     }
-    this.logger.imp(content, this.includeStackTrace);
+    this.log(LogStyle.imp, content);
   }
 
   public warn(content: string | object): void {
-    this.logger.warn(content, this.includeStackTrace);
+    this.log(LogStyle.warn, content);
   }
 
   public err(content: string | object): void {
-    this.logger.err(content, this.includeStackTrace);
+    this.log(LogStyle.err, content);
+  }
+
+  private log(style: LogStyle, content: string | object): void {
+    this.logger[style](content, this.includeStackTrace);
```

Comment:
All the work we had to do to handle when the os file write buffer is full does make me wonder how the Logger implementation is handling when the buffer is full writing to stdout... 🤔

Reply:
Yeah I haven't looked into how console.log works